include/EVT/io/CAN.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef _EVT_CAN_
+#define _EVT_CAN_
+
+#include <stdint.h>
+
+#include <EVT/io/types/CANMessage.hpp>
+
+namespace EVT::core::IO
+{
+// Forward declarations:
+// The different pins are hardware specific. Forward declaration to allow
+// at compilation time the decision of which pins should be used.
+enum class Pin;
+
+/**
+ * Generic interface for CAN bus communication. Devices can send and recieve
+ * messages over the CAN bus using this interface.
+ *
+ * This provides the standard set of features supported across all hardware
+ * for CAN communication. Any hardware specific features are not included.
+ * This interface should be used for the majority of cases and should be
+ * deviated from when there is a need for hardware spoecific functionality.
+ *
+ * NOTE: You cannot directly make an instance of this class via a constructor.
+ * You will need to use the getCAN function in the IO namespace.
+ */
+class CAN
+{
+public:
+    /**
+     * Creates a new instance of the CAN interface which will use the given
+     * transmit and receive pins.
+     *
+     * @param txPin The pin to use for transmitting data
+     * @param rxPin The pin to use for receiving data
+     * @param loopbackEnabled Flag to enable CAN loop back functionality
+     */
+    CAN(Pin txPin, Pin rxPin, bool loopbackEnabled);
```

Comment:
Question, since I'm not 100% on your new architecture layout - is the only way to instantiate the CAN class by means of the manager? If so this is fine, otherwise I'd say put the default option for loopbackEnabled here, and have it false.

Reply:
No the user should be able to use the manager or construct a CAN instance directly, so I agree this should also have a default option.